### PR TITLE
Use a `& 'static` ref of settings rather than cloning configuration at runtime across modules

### PR DESCRIPTION
**Pull request by yvan-sraka**
There are a lot of changes here move around! To get what this PR does you can start by looking at changes made in  `massa-node`!

### CODE REVIEW

**Comment by yvan-sraka** [comment]
A follow-up could be to also read configuration settings this way in `consensus` module :)

File: massa-node/src/main.rs [hunk 1]
```diff
@@ -182,8 +181,66 @@ async fn launch(
...
+        .with_max_level(match SETTINGS.logging.level {
+            4 => Level::TRACE,
+            3 => Level::DEBUG,
+            2 => Level::INFO,
+            1 => Level::WARN,
+            _ => Level::ERROR,
+        })
+        .init();
+
+    // run
```
Comment (AureliaDolo) [inline massa-node/src/main.rs:243]: Why did you move the run part outside of its function ?

**Review by AureliaDolo — COMMENTED** [review]
Why did you renamed some xxxConfig with xxxSettings ?

File: massa-node/src/main.rs (hunk 1 above)
Comment (yvan-sraka) [reply massa-node/src/main.rs:243]: because having it in a separate function call have no purpose here

**Comment by adrien-zinger** [comment]
> Why did you renamed some xxxConfig with xxxSettings ?

The reason is to keep in mind what is new and what is older.

**Comment by adrien-zinger** [comment]
I have tested to a node and it works!

**Review by adrien-zinger — APPROVED** [review]

**Comment by yvan-sraka** [comment]
bors merge
